Userland/Applications/PixelPaint/ImageEditor.h

```diff
@@ -38,6 +39,12 @@ class ImageEditor final
     bool undo();
     bool redo();
 
+    void add_guide(NonnullRefPtr<Guide> guide) { m_guides.append(guide); }
+    void remove_guide(Guide const& guide)
+    {
+        m_guides.remove_first_matching([&](auto entry) { return entry->orientation() == guide.orientation() && entry->offset() == guide.offset(); });
```

Comment:
You can just lookup the pointer here, unless you expect the refcounted guide object to be copied...?

Reply:
I hope I corrected it to do the right thing now. Pointers are a bit magic...
It works however...

Reply:
Should be perfectly fine, the object won't be deleted, so the pointer is unique in the lifetime of that object.